### DISCUSSION TITLE
Is this project abandoned?

### DISCUSSION THREAD

**Discussion by Patrick010** [transmission-remote-gui/transgui#1420] [General]
I see the last release being from 2019 and no responses to issues or questions. 
Too bad because it's by far the best client.

**Comment by username227** [comment]
> I see the last release being from 2019 and no responses to issues or questions. Too bad because it's by far the best client.

This is the only semi-decent remote client that I have found.......I hope that somebody will keep at it.

**Reply by Patrick010** [reply]
Exactly my point. The original webui is laughable, but transgui has some annoying bugs and needs some additional features (as listed in Issues).

**Reply by username227** [reply]
Yeah. Perhaps if somebody is motivated enough they could pick up the project. but i sadly am not a developer.

**Reply by Patrick010** [reply]
Neither am I. And I see that its written in Pascal, so that can be a thing as well.

**Reply by assholehoff** [reply]
Not only is it written in Pascal, but I have gone through a few of the files and I have yet to see a single comment about anything. I don't know Pascal, I don't know what libraries are called, and I can't say that I am enthusiastic about learning Pascal. 

Right now, the most reasonable option for me is to go through the RPC call spec and write something from scratch in Python/Tk or Swift/SwiftUI. Obviously the latter would be Mac only, but it would probably feel awesome. 

I see on PyPi that there are many transmission RPC modules available, so that is probably where I'll start to tinker.

**Reply by Mrnofish** [reply]
A full-featured multi-platform client would probably get a decent amount of attention, please keep us posted.

**Comment by Mrnofish** [comment]
It's not completely abandoned, but it isn't in the prettiest state either.

BTW a faster answer to questions of this kind can be had by checking the activity page: the latest commits are dated end of January / beginning of February.

For those who are OK with the Transmission BT client, Transmission Web Control offers a sophisticated uTorrent-like experience, feel free to check that out.

However, Transmission itself has been progressing at a glacial pace and there are reports of it falling behind in terms of functionality. Nevertheless, it does remain the go-to client for resource-constrained environments, i.e. low-power headless servers, routers, NAS, etc; Transmission will run on 10 or 15 yo embedded systems and that's not something to be taken for granted.

For everything else, there are other clients that may be worth taking a look, qBitTorrent being one of them among the FOSS offerings.

**Reply by Patrick010** [reply]
Transmission Web Control looks nice on the screenshots, but my Chinese is limited to whats on our local Chinese takeout menu, so no idea how to install it.
Transmission itself needs some work also, but atm they are more worried about how the website looks.
I've been using qBittorrent for years, but I like the client-server set up of Transmission, so I switched. There's also Deluge, but thas not any better than Transmission.
Hopefully things will change soon, its really outdated on all levels.

**Reply by Patrick010** [reply]
Small update:
As I have a Synoligy SAN I ran the installation script and it looks very nice. I do prefer a client tho

**Reply by Mrnofish** [reply]
To make the off-topic as brief as possible, qBitTorrent can also be remote-controlled in the same way as Transmission, and there is a a -nox package that's roughly equivalent to transmission-daemon and can be used on headless servers.

One example of daemonizing, is [this unofficial repo which enables -nox use on OpenWrt](https://github.com/KryptonLee/qBittorrent-openwrt-package).

**Reply by Patrick010** [reply]
Thanks, will take a look

**Comment by qu1ck** [comment]
For anyone looking for a client similar in functionality to Transgui, this may be of interest.

I started a rewrite of this project a while ago and recently released first version

https://github.com/openscopeproject/TrguiNG

Builds for all platforms are available in release page.

I consider it to be in beta: stable, functional and usable but may need some polish here and there. Please report any issues you may find if you try this app.

Here are some features that make it stand out from most clients:

* Multi tabbed interface for concurrent server connections (native app only)
* Torrent creation with fast multi threaded hashing (native app only)
* Powerful torrent filtering options (very similar to Transgui)
* Latest transmission features support: labels, bandwidth groups, sequential download
* Dark and white theme
* Works as windows/linux/mac app or as a web interface served by transmission itself

**Reply by assholehoff** [reply]
Wow! Very nice! I am nowhere near anything releasable. 

(I was trying to chose between Swift/SwiftUI and Python/tkinter, so I naturally ended up just tinkering in Common Lisp "while I am deciding"... 😅 Since Transmission Remote GUI has refused to connect to the daemon since I upgraded to 4.0, I have controlled it through the SBCL REPL while building the library.. 🤪)

Looking forward to trying your remote out!👍🏼

**Reply by lighterowl** [reply]
@assholehoff Would you mind trying a build from https://github.com/xavery/transgui and letting me know if you're able to connect now? Thanks.

**Comment by u2n** [comment]
Certainly not abandoned; check Actions page to see @PeterDaveHello just merged some pull requests a couple weeks ago. Review his other projects to see some exemplary open source work. He's a good guy.

A similar utility is [transmission-remote-gtk](https://github.com/transmission-remote-gtk/transmission-remote-gtk), readily available (apt install transmission-remote-gtk).

**Reply by PeterDaveHello** [reply]
Thank you for bringing this discussion into focus. I wasn't fully aware of the ongoing conversations within this GitHub discussion feature, and I'm grateful for the prompt.

The project is not abandoned, but it's not as active as we'd all like. One of the key challenges is the lack of contributors, compounded by the project's programming language and limited resources. These factors already set a high bar for potential contributors. On top of that, the technical requirements for macOS development add another layer of complexity. It's worth noting that most of our past major contributors were not macOS users.

Over the years, I've invested a lot of effort into this project. However, resources are now more constrained, as we all have other commitments that require our attention, including jobs that sustain us.

Anyone interested in contributing is warmly welcomed. Feel free to explore the code and submit PRs. We'll do our best to review and ensure quality, and consistent contributors could be invited to join the maintainer team. Your understanding and patience are invaluable as we navigate these challenges.

**Reply by Mrnofish** [reply]
> On top of that, the technical requirements for macOS development add another layer of complexity. It's worth noting that most of our past major contributors were not macOS users.

Just a reminder: there have been no Windows releases in close to 5 years.

This project is functionally abandoned, if it cannot get even the most basic bugfixes out to the largest user group for what is tantamount to a geological age in IT.

**Comment by lighterowl** [comment]
Shameless self plug : I do my best to maintain a fork at https://github.com/xavery/transgui , try to address any pressing issues and make semi-regular releases for all supported platforms.

**Comment by Marius8** [comment]
I know, I kind of necro this one..
But the last build is still OLDER THAN COVID..
the fork by lighterowl (former xavery) is now archived, too..

**Comment by perfect7gentleman** [comment]
But there is activity.

**Reply by Marius8** [reply]
activity, yes..
the latest release is still from october 12th 2019

**Reply by username227** [reply]
https://github.com/openscopeproject/TrguiNG

**Reply by PeterDaveHello** [reply]
Will have the new release soon.

**Reply by u2n** [reply]
> Will have the new release soon.

Glad to know it lives, TY and looking forward to. That said, the new gen version linked by @username227 is also appealing and uses more current tools (so easier to attract contributors, keep updated). Will test both, might lead to a long period of indecision :)
